### PR TITLE
Remove ReadOnlyConfiguration

### PR DESCRIPTION
**Pull request by zabulus**
Fixes
https://github.com/MRCollective/ChameleonForms/issues/119

### CODE REVIEW

File: ChameleonForms.Tests/Component/Config/ReadonlyFieldConfigurationTests.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ class ReadonlyFieldConfigurationShould
         [Test]
```
Comment (robdmoore) [inline ChameleonForms.Tests/Component/Config/ReadonlyFieldConfigurationTests.cs]: This whole file can be deleted. It's not needed anymore.

File: ChameleonForms/Component/Config/FieldConfiguration.cs [hunk 2]
```diff
@@ -761,7 +669,7 @@ public string ToHtmlString()
         /// <inheritdoc />
         public IReadonlyFieldConfiguration ToReadonly()
```
Comment (robdmoore) [inline ChameleonForms/Component/Config/FieldConfiguration.cs]: I wonder if we still need this method?

**Comment by robdmoore** [comment]
Given this type was public we will need to document the removal in BREAKING_CHANGES.md - do you mind trying that as part of this PR?

**Comment by zabulus** [comment]
Fixed issues you mentioned. 
Edited Breaking_Changes. Not sure if added record correct, though.

**Comment by robdmoore** [comment]
Thanks for this contribution!

**Pull request merged**